targets/hasura/metadata/tables.yaml

```diff
@@ -176,11 +176,73 @@
       - updated_at
       filter: {}
       check: {}
+- table:
+    schema: public
+    name: document_relations
+  object_relationships:
+  - name: a
+    using:
+      foreign_key_constraint_on: document_a
+  - name: b
+    using:
+      foreign_key_constraint_on: document_b
+  select_permissions:
+  - role: public
+    permission:
+      columns:
+      - id
+      - document_a
+      - document_b
+      - type
+      - data
+      filter: {}
+      allow_aggregations: true
+  - role: user
+    permission:
+      columns:
+      - id
+      - document_a
+      - document_b
+      - type
+      - data
+      filter: {}
+      allow_aggregations: true
 - table:
     schema: public
     name: documents
+  array_relationships:
+  - name: relation_a
+    using:
+      foreign_key_constraint_on:
+        column: document_a
+        table:
+          schema: public
+          name: document_relations
+  - name: relation_b
```

Comment:
not sur to understand why we need 2 relations here

Reply:
We probably don't ! Nice catch. I need to think about it. It was suggested by hasura but if think I can indeed remove one

Reply:
I can't :/
Let's say I have a theme in the middle of my graph/tree. I want both the parent, and the children. To get the parent, I need to pick the relation where my current theme's id is exclusively in the `document_b` column (relation_b) and then get `document_a` of this relation row. If I ever tried to find the parent by fetching relations where the theme id is in column `document_a`, I'm in a dead end becaue I could then only link it to their document_b which can only be children.

Reply:
They are two alternative:
1- Let say we have theme A - B and C. If I want to fetch theme B with its parent (A) and child (C) I could double all the relations so that both A and C would end up in the first column of the relationship with a property saying if it's a parent / children relation. But that means doubling all the relation since any relation can be both parent and children. Quite pain
2 - I could make several queries instead of one and don't use the relashionships provided by hasura but that would quite a pain too.